nilearn/plotting/displays.py

```diff
@@ -640,9 +640,12 @@ def _map_show(self, img, type='imshow',
             else:
                 data = np.ma.masked_inside(data, -threshold, threshold,
                                            copy=False)
-            img = new_img_like(img, data, img.get_affine())
+            img = new_img_like(img, data, _utils.compat.get_affine(img))
 
-        affine = img.get_affine()
+        if img.__class__.__name__ == '_MNI152Template':
+            affine = img.get_affine()
+        else:
+            affine = _utils.compat.get_affine(img)
```

Comment:
The code that you have above is a bit hacky. A better way to solve this issue would be to fix the "affine" attribute of the _MNI152Template. The way you can do this is using Python "properties", they are functions that are called upon access of an attribute. It is actually done in the _MNI152Template for the 'shape' attribute, so you have to mimic the strategy used for "shape".